.github/workflows/lint.yaml

```diff
@@ -0,0 +1,45 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [lts/*]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup node env
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Install foundry-toolchain
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Run foundry node, deploy contracts (& generate contracts typescript output)
+        run: yarn fork & yarn deploy:factory
```

Comment:
Ideally we would execute these commands as part of CI pipeline to ensure nothing is broken by a PR, but it depends on the following env vars:

```
DEPLOYER_PRIVATE_KEY=
SEPOLIA_RPC_URL=
```

So need to set them as repository secrets in order for this to work. Happy to volunteer one of my burner testnet only wallets to see if my hypothesis is correct